app/shared-components/PromoModal/PromoModal.jsx

```diff
@@ -0,0 +1,201 @@
+import PropTypes from 'prop-types';
+import ClassNames from 'classnames';
+import Modal from '../Modal';
+import InsightsPromoModalContent from '../ModalContent/InsightsPromoModalContent';
+import PlusPromoModalContent from '../ModalContent/PlusPromoModalContent';
+import PremiumPromoModalContent from '../ModalContent/PremiumPromoModalContent';
+import history from '../../panel/utils/history';
+import { sendMessage } from '../../panel/utils/msg';
+import globals from '../../../src/classes/Globals';
+import ModalExitButton from '../../panel/components/BuildingBlocks/ModalExitButton';
+
+const DOMAIN = globals.DEBUG ? 'ghosterystage' : 'ghostery';
+const INSIGHTS = 'insights';
+const PLUS = 'plus';
+const PREMIUM = 'premium';
+
+/**
+ * A base functional component for Promo Modals
+ * @return {JSX}
+ * @memberof HubComponents
+ */
+
+class PromoModal extends React.Component {
+	/**
+	 * @param modal			'insights' or 'premium'
+	 * @private
+	 * Handle clicks on the link to turn off promos in the promo modals
+	 */
+	_handlePromoGoAwayClick = (modal) => {
+		this.props.actions.togglePromoModal();
+
+		sendMessage('promoModals.turnOffPromos', {});
+
+		switch (modal) {
+			case INSIGHTS:
+				sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+				break;
+			case PLUS:
+				sendMessage('ping', 'promo_modals_decline_plus_upgrade');
+				break;
+			case PREMIUM:
+				sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+				break;
+			default:
+				break;
+		}
+
+		this.props.actions.showNotification({
+			classes: 'warning',
+			reload: false,
+			text: t('promos_turned_off_notification'),
+		});
+	};
+
+	/**
+	 * @private
+	 * Handle clicks on the download buttons
+	 */
+	_handlePromoTryProductClick = (product, utm_campaign) => {
+		this.props.actions.togglePromoModal();
+
+		let url;
+		switch (product) {
+			case PLUS:
+				// Should we send a plus ping here?
+				// sendMessage('ping', 'promo_modals_plus_upgrade_cta');
+				url = `https://checkout.${DOMAIN}.com/plus?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			case PREMIUM:
+				url = `https://ghostery.com/thanks-for-downloading-midnight?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			case INSIGHTS:
+				sendMessage('ping', 'promo_modals_insights_upgrade_cta');
+				url = `https://checkout.${DOMAIN}.com/insights?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			default:
+		}
+
+		sendMessage('openNewTab', {
+			url,
+			become_active: true,
+		});
+	}
+
+	/**
+	 * @private
+	 * Handle clicks on sign in links in promo modals
+	 */
+	_handlePromoSignInClick = () => {
+		this.props.actions.togglePromoModal();
+		history.push({
+			pathname: '/login',
+		});
+	};
+
+	_handlePromoXClick = (type) => {
+		this.props.actions.togglePromoModal();
+
+		if (type === INSIGHTS) {
+			sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+		} else if (type === PLUS) {
+			sendMessage('ping', 'promo_modals_decline_plus_upgrade');
+		}
```

Comment:
Do we need to send a decline ping for Premium too?

Reply:
Same as above